src/Microsoft.Health.Dicom.Core/Features/Common/DicomTagParser.cs

```diff
@@ -27,24 +28,79 @@ public bool TryParse(string dicomTagPath, out DicomTag[] dicomTags, bool support
                 return false;
             }
 
+            DicomTag dicomTag = GetTagFromKeywordOrNumber(dicomTagPath);
 
+            dicomTags = new DicomTag[] { dicomTag };
+            return dicomTag != null;
+        }
+
+        public bool TryParseToDicomItem(string dicomTagPath, out DicomItem dicomItem)
```

Comment:
What value are you getting from returing the DicomItem? Is'nt it implicit that the more than one tag returned is a SQ?

Can we change the above TryParse(supportMultiple = true)? for WorkItem? and then later for DicomInstance?

You have one method that handles all cases

Reply:
We are getting a DICOM sequence out, which is a dicom item. Dicom sequence is not just list of tags but list of datasets

Reply:
Changed as per our discussion, now it returns multiple tags for sequence tag path. We removed supportMultiple flag to keep it simple and it wasnt necessary now